src/components/Form.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import React, { useState } from 'react';
+
+type FormProps = {
+  onSubmit: (value: number) => void;
+};
+
+export default function Form({ onSubmit }: FormProps) {
+  const [inputValue, setInputValue] = useState('');
+  const [error, setError] = useState('');
+  const isDisabled = inputValue === '';
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    const sanitizedValue = value.replace(/\D/g, '');
+
+    if (sanitizedValue === '') {
+      setInputValue('');
+      setError('');
+      return;
+    }
```

Comment:
[imo]
今の所、数字以外の値を入力すると何も表示されずユーザーが混乱しそうなため、案内はあった方が良さそうです
```suggestion
    const sanitizedValue = value.replace(/\D/g, '');

    if (sanitizedValue === '') {
      setInputValue('');
      setError('半角数字を入力してください');
      return;
    }
```